src/pages/diagram/dialogs/edit-diagram/edit-diagram.tsx

```diff
@@ -1,58 +1,96 @@
-import { useState } from "react";
-
 import { edit } from "services/resources/diagrams/edit";
-import { IEdit } from "./types/IEdit";
 
 import { useDiagram } from "pages/diagram/hooks/useDiagram";
 
+import jwt_decode from "jwt-decode";
+import { SessionStorage } from "shared/utils/base-storage/session-storage";
+
 import { useSnackbar } from "notistack";
 
 import * as S from "./styles";
-
+import { create } from "services/resources/diagrams/create";
+import { useParams } from "react-router-dom";
+import { TUser } from "models/user";
 
 type Props = {
   isOpen: boolean;
   onClose?: () => void;
   onClick?: React.MouseEventHandler<HTMLButtonElement>;
   id: string;
+  xml: string;
 };
 
 export const EditDiagram: React.FC<Props> = ({ isOpen, onClose, id }) => {

```

Comment:
Já que este component será utilizado tanto para criar um diagrama novo quanto para editar um diagrama já existente, seria mais interessante alterar o nome. Exemplo: EditDiagram para FormDiagram.

Obs.: a sugestão de nome foi feita apenas para exemplificar, fique a vontade pra pensar em um outro nome :)

```suggestion
export const FormDiagram: React.FC<Props> = ({ isOpen, onClose, id }) => {

```